awscli/customizations/history/list.py

```diff
@@ -0,0 +1,499 @@
+        #     no next element, or the next element is another param.
+        # In case 1 we will move the param down a line even though it
+        # would fit by itself, this gives it a better chance of sharing
+        # a line with its value, even if the value doesn't fit it will
+        # be on the next line slightly indented making it easy to
+        # identifiy.
+        # In case 2 and 3 we will leave the paramater on the current
+        # line since there is nothing to try and pair it with.
+        next_line, next_space = self._append_value(
+            line, candidate, space)
+        next_element = self._peek()
+        if next_element is None or self._is_param(next_element):
+            # Handle case 3. Do nothing since it will be included by
+            # default in the if fits block below.
+            return False
+        elif next_element is not None and self._does_fit_on_line(
+                next_line, next_element, next_space):
+            # case 2 the next element fits on this line as well so
+            # we can leave the current element on this line.
+            return False
+        else:
+            # Neither of the above are true so this must be case 1 where
+            # the next element existed and did not fit on the current row.
+            # So we will put this param element back into the list stop
+            # formatting this row by breaking out of the loop.
+            return True
+
+    def _indent_line_if_needed(self, line):
+        if not self._indented:
+            self._indent()
+            return line
+        else:
+            return '%s%s' % (self._NON_FIRST_LINE_INDENTATION, line)
+
+    def _indent_candidate(self, line, candidate, param):
+        # If it is not a param and not starting a line then it is
+        # probably a paramater value that could not fit on the same
+        # line as its parameter so we will indent it a little extra
+        # for clarity.
+        if self._indented and not line and not param and \
+           self._last_line_ended_in_param:
+            candidate = '%s%s' % (self._PARAM_VALUE_INDENTATION, candidate)
+        return candidate
+
+    def _truncate_and_add_to_line(self, line, candidate, param, space):
+        truncated_candidate = self._truncate_to_fit(
+            candidate, space)
+        line, space = self._add_to_line(
+            line, truncated_candidate, param, space)
+        return line, space
+
+    def _end_line_early(self, line, candidate, param, fits, space):
+        # If candidate is a param and there is already content on this
+        # line and the parameter would fit we need to decide if we are
+        # going to skip it. If we do we need to put it back in the value
+        # so it can be
+        if param and line and fits:
+            if self._skip_param(line, candidate, space):
+                self._value.insert(0, candidate)
+                return True
+        return False
+
+    def get_line(self):
```

Comment:
I don't think this accounts for newlines in the values themselves which will mess up the format, e.g. an argument with a json value.